autorest/codegen/__init__.py

```diff
@@ -306,8 +327,16 @@ def _build_code_model_options(self) -> Dict[str, Any]:
         if azure_arm:
             options["credential"] = True
             options["head_as_boolean"] = True
+
+        if options["package_mode"] == 'dataplane':
```

Comment:
I don't think adding these options is in the scope of this PR, not sure all of these should default to these values as well

Reply:
Correct, the scope of the PR is templating, not changing behavior

Reply:
Usually we don't advise users to use `package-mode` with readme.md. But actually we can't. So let me think about more here.

Reply:
Usually, `package_mode` is not advised to use together with readme.md so it will not have bad effect. But actually we can't stop that so let me think about more here.

Reply:
The changing behavior is necessary to generate a whole package: `package-mode` only asks users to input limited parameter. So it is necessary to help configure some other parameters. 
For example, codegen asks to input  `--credential` if users want to use `--credential-scopes`. For convenience, we want users to input limited parameters. So `package-mode` only asks to input `--credential-scopes`. Then we have to help to replenish `options["credential"] = True`